src/gt4py/next/iterator/transforms/inline_lambdas.py

```diff
@@ -63,53 +66,24 @@ def inline_lambda(  # see todo above
     if node.fun.params and not any(eligible_params):
         return node
 
-    refs = set().union(
-        *(
-            arg.pre_walk_values().if_isinstance(ir.SymRef).getattr("id").to_set()
-            for arg, eligible in zip(node.args, eligible_params)
-            if eligible
-        )
-    )
-    syms = node.fun.expr.pre_walk_values().if_isinstance(ir.Sym).getattr("id").to_set()
-    clashes = refs & syms
-    expr = node.fun.expr
-    if clashes:
-        # TODO(tehrengruber): find a better way of generating new symbols in `name_map` that don't collide with each other. E.g. this must still work:
-        # (lambda arg, arg_: (lambda arg_: ...)(arg))(a, b)  # noqa: ERA001 [commented-out-code]
-        name_map: dict[ir.SymRef, str] = {}
-
-        def new_name(name):
-            while name in refs or name in syms or name in name_map.values():
-                name += "_"
-            return name
-
-        for sym in clashes:
-            name_map[sym] = new_name(sym)
-
-        expr = RenameSymbols().visit(expr, name_map=name_map)
-
     symbol_map = {
```

Comment:
Just leaving a remark here, that there was #2134. Maybe it's already resolved here.